client/lib/route/add-query-args.js

```diff
@@ -12,6 +12,14 @@ export default function( args, url ) {
 		throw new Error( 'addQueryArgs expects the second argument to be a string.' );
 	}
 
+	// Remove any undefined query args
+	args = Object.keys( args ).reduce( ( defined, key ) => {
```

Comment:
I think if one were to treat `addQueryArgs` as a glorified [`querystring.stringify`](https://nodejs.org/api/querystring.html#querystring_querystring_stringify_obj_sep_eq_options), this could cause some confusion if they were aware of the caveats and treatment around assigning `null` or `undefined` values. In fact, came up a bit recently at https://github.com/Automattic/wp-calypso/pull/6468#issuecomment-231267271 (using `null` intentionally, cc @timmyc).

That said, I don't think we should necessarily treat it this way, and I like the idea of conveniences for sane treatment of "empty" parameters.

Only thing I might suggest is considering [`_.pickBy`](https://lodash.com/docs#pickBy) to simplify this a bit:

```js
args = pickBy( args, ( arg ) => arg != null );
```

Non-strict-equals here is intentional [and permitted](https://github.com/Automattic/eslint-config-wpcalypso/blob/271bdfee6ad3c946d9ddf277d7a362371a1d6f6b/index.js#L25). Not entirely sure I like it, but this is the exact use case for it ([ESLint docs explain](http://eslint.org/docs/rules/eqeqeq#allow-null)).